ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -433,6 +438,23 @@ public synchronized void storeBlock(final Block block, final List<TransactionRec
     appendBlockHelper(new BlockWithReceipts(block, receipts), true, true);
   }
 
+  @Override
+  public void unsafeStoreHeader(
+      final BlockHeader blockHeader,
+      final Difficulty totalDifficulty,
+      final boolean updateChainHead) {
+    final BlockchainStorage.Updater updater = blockchainStorage.updater();
+    updater.putBlockHeader(blockHeader.getHash(), blockHeader);
+    updater.putBlockHash(blockHeader.getNumber(), blockHeader.getBlockHash());
+    updater.putTotalDifficulty(blockHeader.getHash(), totalDifficulty);
```

Comment:
totalDifficulty was removed in the subcommand feature: https://github.com/hyperledger/besu/pull/8388/files#diff-414a8263b2b105fcb93e3cff4c23e8fca82a2befb39a84d8a3a1c553ccfd4486R130

Is totalDifficultly needed for syncing?

Reply:
discussed that removing it would break things and want to keep this PR simple. May look at removing totalDifficultly for sync path later